### ISSUE TITLE
stanza for React component / react で stanza を使いたい

### ISSUE THREAD

**Issue by Ningensei848** [togostanza/metastanza#27]
I want to use togostanza / metastanza in a way that partially rewrites the virtual DOM like react. However, as you can see from the `querySelector ()` method etc., **the DOM elements are referenced directly from JS**. I don't like this very much.

 I also understand your idea that the ease of copying and pasting in HTML is important. But as it stands, I have a feeling that this project is going against modern web development that dynamically generates content.

 After all, **I want to combine a common text search form with a stanza**, but I find it difficult for developers to easily create in a modern way.I would appreciate it if you could consider improving it.

P.S.
 Or it may be my misunderstanding or lack of understanding. Would you please teach me a good way?

---

　`React.js` のように仮想DOMを部分的に書き換えるような方法で togostanza / metastanza を使いたいです．しかし，`querySelector()` メソッドなどを見ればわかるように，ページのDOM要素をJSから直接参照しています．これはあまり好きではありません．

　コピペしてHTMLに貼るだけという手軽さこそが大事というあなた方の思想も理解しています．しかし今のままでは，このプロジェクトはコンテンツを動的に生成するモダンなウェブ開発と逆行している予感もします．

　最終的には，普通の一般的なテキスト検索フォームとstanzaを組み合わせたい（テキスト入力で `stanza` の属性値を動的に変更したい）だけなのですが，いまのままだと開発者に一工夫必要で難しいと感じています．改善をご検討いただければ幸いです．

P.S. 
もしくは私の勘違いか或いは理解が不足しているということも考えられます．良いやり方をご教授いただけませんか？

**Comment by darashi** [comment]
I completely agree with you. We're trying to integrate togostanza with the modern web frontend frameworks based on virtual DOM.

We’ve implemented a mechanism that enables us to hook into the build process of the stanzas (not documented yet because the specification has not been finalized), and with this mechanism, we actually use Vue.js to implement a stanza. We haven't actually tried but if we do something similar to this, we're thinking React could be used as well.

This is the stanza:

https://github.com/togostanza/metastanza/blob/master/stanzas/table-with-pagination/index.js

Build configuration for Vue.js:

https://github.com/togostanza/metastanza/blob/master/togostanza-build.mjs

In the current implementation, the stanza is completly rewritten every time the attribute changed. We understand that this is not desirable behaviour when combied with such modern frameworks. We should need to provide an API that enables partial redrawing. We also looking into this issue as well.

Is that the answer to your question?

----

おっしゃるとおりだと思います。私たちも、 Virtual DOM を利用したモダンな Web フロントエンドフレームワークと togostanza とを統合できるようにしようと試みているところです。

スタンザをビルドするプロセスにフックする仕組みを実装しており（これはまだ仕様が確定していないためドキュメントされていません）、これを利用して実際に Vue.js を用いてスタンザを実装しています。実際に試したわけではないのですが、これと同じようなことをすれば、React も利用できるのではないでしょうか。

(URL略)

現在の実装では、属性値が書き換えられるたびにスタンザが完全に再描画されてしまいます。この挙動は、モダンなフレームワークと組み合わせるときに問題になると思います。部分的な再描画を行うAPIを提供する必要がありそうです。これについても検討しているところです。

これで回答になっているでしょうか？

**Comment by Ningensei848** [comment]
Thank you very much for your prompt reply. 😃 

 When looking over the project, I occasionally saw  `vue.js` -like code.  You say this is not yet documented, I would wait until the specification is finalised.

 Also, I'm not sure if I have a deep understanding of the specs of this project, but I'll try to implement it in `React.js` using the examples. I would like to send you a PR if it is possible ... Thank you very much for your answer. 👍 

---

　迅速なご回答をいただき，深く感謝いたします．所々で見かけた `vue.js` らしきコードはそういう意図があったのですね．例まで示していただき，ありがたく存じます．仕様が定まるのをお待ちしております．

　まだまだ理解が浅い部分はあるかと思いますが，例を参考に `React.js` でも実現できないか試してみます．出来そうならPRを送ろうかとおもいます．ご回答ありがとうございました～
